cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go

```diff
@@ -469,78 +464,83 @@ func (s *Server) createRepoKubeappsManagedSecret(
 // see https://github.com/vmware-tanzu/kubeapps/pull/4630#discussion_r861446394 for details
 func (s *Server) setOwnerReferencesForRepoSecret(
 	ctx context.Context,
+	secret *apiv1.Secret,
 	repo *sourcev1.HelmRepository) error {
 
-	if repo.Spec.SecretRef != nil {
+	if repo.Spec.SecretRef != nil && secret != nil {
 		if typedClient, err := s.clientGetter.Typed(ctx, s.kubeappsCluster); err != nil {
 			return err
 		} else {
 			secretsInterface := typedClient.CoreV1().Secrets(repo.Namespace)
-			if secret, err := secretsInterface.Get(ctx, repo.Spec.SecretRef.Name, metav1.GetOptions{}); err != nil {
-				return statuserror.FromK8sError("get", "secrets", repo.Spec.SecretRef.Name, err)
-			} else {
-				secret.OwnerReferences = []metav1.OwnerReference{
-					*metav1.NewControllerRef(
-						repo,
-						schema.GroupVersionKind{
-							Group:   sourcev1.GroupVersion.Group,
-							Version: sourcev1.GroupVersion.Version,
-							Kind:    sourcev1.HelmRepositoryKind,
-						}),
-				}
-				if _, err := secretsInterface.Update(ctx, secret, metav1.UpdateOptions{}); err != nil {
-					return statuserror.FromK8sError("update", "secrets", repo.Spec.SecretRef.Name, err)
-				}
+			secret.OwnerReferences = []metav1.OwnerReference{
+				*metav1.NewControllerRef(
+					repo,
+					schema.GroupVersionKind{
+						Group:   sourcev1.GroupVersion.Group,
+						Version: sourcev1.GroupVersion.Version,
+						Kind:    sourcev1.HelmRepositoryKind,
+					}),
+			}
+			if _, err := secretsInterface.Update(ctx, secret, metav1.UpdateOptions{}); err != nil {
+				return statuserror.FromK8sError("update", "secrets", secret.Name, err)
```

Comment:
Nice - so the main change to this file is just the return type of `validateUserManagedRepoSecret` and so that the arg to `setOwnerReferencesForRepoSecret` can include the secret (and avoid an extra lookup?)

Reply:
> Nice - so the main change to this file is just the return type of `validateUserManagedRepoSecret` and so that the arg to `setOwnerReferencesForRepoSecret` can include the secret (and avoid an extra lookup?)

Well, I would say the 'redacted' string handling, is the main change to this file. What you mention is just a small unrelated performance improvement, an icing on the cake, so to speak.

Thanks for the review
Greg